### PR TITLE
docker tag management

### PR DESCRIPTION
**Pull request by nicolasochem**
So far, we have only been pushing the "latest" tag.

We need to tag every docker image differently (release tags, commit tags)
to prevent unwanted upgrades of trd container.

We will still have a "latest" tag but we wouldn't recommend using it.

We are using the method described in github documentation:

https://docs.github.com/en/actions/guides/publishing-docker-images#publishing-images-to-docker-hub

### CODE REVIEW

**Comment by jdsika** [comment]
@vkresch I am not experienced with docker. Explanation makes sense to me but could you approve, please?

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -21,8 +21,14 @@ jobs:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
 
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: trdo/tezos-reward-distributor
       - name: Push to Docker Hub
```
Comment (vkresch) [inline .github/workflows/docker.yml:29]: Add one blank line above just for aesthetic reason ;)

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -21,8 +21,14 @@ jobs:
...
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v5
```
Comment (vkresch) [inline .github/workflows/docker.yml:30]: Does docker/build-push-action has already a version 5? On their [github page](https://github.com/docker/build-push-action) they still use v2 (see [example](https://github.com/docker/build-push-action/blob/master/docs/advanced/tags-labels.md)). Or am I missing something?

**Review by vkresch — DISMISSED** [review]
Looks good. Does docker/build-push-action has a version 5?

File: .github/workflows/docker.yml (hunk 2 above)
Comment (nicolasochem) [reply .github/workflows/docker.yml]: My bad, I was looking at the wrong action.

**Review by jdsika — APPROVED** [review]

**Pull request merged**